scipy/stats/stats.py

```diff
@@ -601,7 +623,7 @@ def hmean(a, axis=0, dtype=None):
         raise ValueError("Harmonic mean only defined if all elements greater than zero")
 
 
-def mode(a, axis=0):
+def mode(a, axis=0, nan_policy='propagate'):
```

Comment:
Why is that backwards compatible? ``'propagate'`` should do the right thing here, if you let it actually propagate (note that that doesn't mean "return nans" in this case).